images/hub/jupyterhub_config.py

```diff
@@ -126,8 +126,12 @@
 c.KubeSpawner.mem_guarantee = get_config('singleuser.memory.guarantee')
 c.KubeSpawner.cpu_limit = get_config('singleuser.cpu.limit')
 c.KubeSpawner.cpu_guarantee = get_config('singleuser.cpu.guarantee')
-c.KubeSpawner.extra_resource_limits = get_config('singleuser.extra_resource.limits')
-c.KubeSpawner.extra_resource_guarantees = get_config('singleuser.extra_resource.guarantees')
+extra_resource_limits = get_config('singleuser.extra_resource.limits')
+if extra_resource_limits:
+    c.KubeSpawner.extra_resource_limits = extra_resource_limits
+extra_resource_guarantees =  get_config('singleuser.extra_resource.guarantees')
+if extra_resource_guarantees:
+    c.KubeSpawner.extra_resource_guarantees = extra_resource_guarantees
```

Comment:
```
c.KubeSpawner.extra_resource_limits = get_config('singleuser.extra-resource.limits', {})
c.KubeSpawner.extra_resource_guarantees = get_config('singleuser.extra-resource.guarantees', {})
```

Reply:
I thought about doing it this way, then decided it was safer to not set it at all so that if the KubeSpawner.* defaults change in future this will "just work". This is the pattern used for e.g. `lifecycle_hooks`

Reply:
Hmmm, yes, I guess that would alllow KubeSpawner to add a resource field that would propegate to all z2jh deployment not specifying an extra resource of their own.

But if we want to do this, we should extend the value of c.KubeSpawner.extra_resource_garantees rather than override them when we do have a value.

---

I considered `lifecycle_hooks` as well and other fields like `cmd` and wasn't very happy about consistency of how we do things in `jupyterhub_config.py` of the z2jh chart.

I can consider this later systematically throughout the config instead.